File: frontend/src/types/blog.ts
export interface Blog {
  id: number;
  title: string;
  slug: string;
  excerpt?: string;
  content: string;
  featuredImageUrl?: string;
  featuredImageAlt?: string;
  status: 'draft' | 'published' | 'archived';
  authorId: string;
  authorName?: string;
  tags?: string;
  category?: string;
  viewCount: number;
  likeCount: number;
  publishedAt?: string;
  createdAt: string;
  updatedAt: string;
  metaTitle?: string;
  metaDescription?: string;
  metaKeywords?: string;
}

export interface CreateBlogRequest {
  title: string;
  excerpt?: string;
  content: string;
  featuredImageUrl?: string;
  featuredImageAlt?: string;
  status: 'draft' | 'published' | 'archived';
  tags?: string;
  category?: string;
  metaTitle?: string;
  metaDescription?: string;
  metaKeywords?: string;
}

export interface UpdateBlogRequest {
  title: string;
  excerpt?: string;
  content: string;
  featuredImageUrl?: string;
  featuredImageAlt?: string;
  status: 'draft' | 'published' | 'archived';
  tags?: string;
  category?: string;
  metaTitle?: string;
  metaDescription?: string;
  metaKeywords?: string;
}

export interface BlogListResponse {
  blogs: Blog[];
  totalCount: number;
  page: number;
  pageSize: number;
}

export interface BlogFilters {
  status?: string;
  category?: string;
  authorId?: string;
  search?: string;
  page?: number;
  pageSize?: number;
}

File: backend/Models/Blog.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models
{
    public class Blog
    {
        public int Id { get; set; }
        
        [Required]
        [MaxLength(200)]
        public string Title { get; set; } = string.Empty;
        
        [Required]
        [MaxLength(500)]
        public string Slug { get; set; } = string.Empty;
        
        [MaxLength(1000)]
        public string? Excerpt { get; set; }
        
        [Required]
        public string Content { get; set; } = string.Empty;
        
        [MaxLength(500)]
        public string? FeaturedImageUrl { get; set; }
        
        [MaxLength(100)]
        public string? FeaturedImageAlt { get; set; }
        
        [Required]
        [MaxLength(50)]
        public string Status { get; set; } = "draft"; // draft, published, archived
        
        [Required]
        public string AuthorId { get; set; } = string.Empty;
        
        [MaxLength(100)]
        public string? AuthorName { get; set; }
        
        [MaxLength(500)]
        public string? Tags { get; set; } // Comma-separated tags
        
        [MaxLength(100)]
        public string? Category { get; set; }
        
        public int ViewCount { get; set; } = 0;
        
        public int LikeCount { get; set; } = 0;
        
        public DateTime? PublishedAt { get; set; }
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
        
        // SEO fields
        [MaxLength(200)]
        public string? MetaTitle { get; set; }
        
        [MaxLength(500)]
        public string? MetaDescription { get; set; }
        
        [MaxLength(200)]
        public string? MetaKeywords { get; set; }
        
        // Navigation properties
        [ForeignKey("AuthorId")]
        public virtual User Author { get; set; } = null!;
    }
}
